server/mailet/rate-limiter/src/main/scala/org/apache/james/transport/mailets/PerSenderRateLimit.scala

```diff
@@ -0,0 +1,220 @@
+  override def asString(): String = keyPrefix.map(prefix => s"${prefix.value}_${entityType.asString()}_${mailAddress.asString()}")
+    .getOrElse(s"${entityType.asString()}_${mailAddress.asString()}")
+}
+
+/**
+ * <p><b>PerSenderRateLimit</b> allows defining and enforcing rate limits for the sender of matching emails.</p>
+ *
+ * <ul>This allows writing rules like:
+ *   <li>A sender can send 10 emails per hour</li>
+ *   <li>A sender can send email to a total of 20 recipients per hour</li>
+ *   <li>A sender can send 100 MB of emails (total computed taking only the email size into account) per hour</li>
+ *   <li>A sender can send 200 MB of emails (total computed taking each recipient copies into account) per hour</li>
+ * </ul>
+ *
+ * <p>Depending on its position and the matcher it is being combined with, those rate limiting rules could be applied to
+ * submitted emails, received emails or emitted email being relayed to third parties.</p>
+ *
+ *  <ul>Here are supported configuration parameters:
+ *    <li><b>keyPrefix</b>: An optional key prefix to apply to rate limiting. Choose distinct values if you specify
+ *    this mailet twice within your <code>mailetcontainer.xml</code> file. Defaults to none.</li>
+ *    <li><b>exceededProcessor</b>: Processor to which emails whose rate is exceeded should be redirected to. Defaults to error.
+ *    Use this to customize the behaviour upon exceeded rate.</li>
+ *    <li><b>duration</b>: Duration during which the rate limiting shall be applied. Compulsory, must be a valid duration of at least one second. Supported units includes s (second), m (minute), h (hour), d (day).</li>
+ *    <li><b>count</b>: Count of emails allowed for a given sender during duration. Optional, if unspecified this rate limit is not applied.</li>
+ *    <li><b>recipients</b>: Count of recipients allowed for a given sender during duration. Optional, if unspecified this rate limit is not applied.</li>
+ *    <li><b>size</b>: Size of emails allowed for a given sender during duration (each email count one time, regardless of recipient count). Optional, if unspecified this rate limit is not applied. Supported units : B ( 2^0 ), K ( 2^10 ), M ( 2^20 ), G ( 2^30 ), defaults to B.</li>
+ *    <li><b>totalSize</b>: Size of emails allowed for a given sender during duration (each recipient of the email email count one time). Optional, if unspecified this rate limit is not applied. Supported units : B ( 2^0 ), K ( 2^10 ), M ( 2^20 ), G ( 2^30 ), defaults to B. Note that
+ *    totalSize is limited in increments of 2exp(31) - ~2 billions: sending a 10MB file to more than 205 recipients will be rejected if this parameter is enabled.</li>
+ *  </ul>
+ *
+ *  <p>For instance, to apply all the examples given above:</p>
+ *
+ *   <pre><code>
+ * &lt;mailet matcher=&quot;All&quot; class=&quot;PerSenderRateLimit&quot;&gt;
+ *     &lt;keyPrefix&gt;myPrefix&lt;/keyPrefix&gt;
```

Comment:
format...difficult to read, 
is this just temporary?

Reply:
Nope, javadoc needs escaped HTML! That's needed for the mailet doc plugin...  No choice ;-) Hopefully the readme is more friendly.